glusterd2/commands/volumes/volume-create.go

```diff
@@ -42,6 +46,10 @@ func validateVolCreateReq(req *api.VolCreateReq) error {
 			return gderrors.ErrEmptyBrickList
 		}
 	}
+	metadataSize := volume.GetMetadataLen(req.Metadata)
```

Comment:
Same comment here. The method should on the type and not in volume package. Name the method as `MetadataSize` and the call will look like:

```go
if req.MetadataSize() > maxMetadataSizeLimit {
```